app/src/main/java/com/poupa/vinylmusicplayer/util/DataTypeUtil.java

```diff
@@ -0,0 +1,29 @@
+package com.poupa.vinylmusicplayer.util;
+
+import com.poupa.vinylmusicplayer.model.CategoryInfo;
+
+import java.lang.reflect.Type;
+
+public class DataTypeUtil {
```

Comment:
Can we avoid this class altogether by saving prefs in json (where data type is preserved) instead of saving in ini (where datatype is lost)?

Reply:
If we go with json, it should be`jsonc` (commented json)

Reply:
> Can we avoid this class altogether by saving prefs in json (where data type is preserved) instead of saving in ini (where datatype is lost)?

I don't have any experience with reading in json with java, but I can try it. I also think that guessing the data type is not the best approach.

Reply:
I would rely on existing library to do the read/write. Looks like Gson does the job, except it doesnt support comments.

Reply:
I've done this.